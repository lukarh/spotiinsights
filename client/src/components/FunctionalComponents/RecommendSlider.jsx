import { Box, Slider } from "@mui/material"

const RecommendSlider = ({ ariaLabel, value, min, max, step, handleChange, color }) => {

    return (
        <Box>
            <p>{ariaLabel}</p>
            <Slider
                data-testid="recommend-slider"
                value={value}
                onChange={handleChange}
                min={min}
                max={max}
                step={step}
                valueLabelDisplay="auto"
                aria-labelledby="range-slider"
                style={{ color: color }}
            />
        </Box>
    )
}

export default RecommendSlider;