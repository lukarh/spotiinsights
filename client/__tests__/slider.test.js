import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render, cleanup, screen } from "@testing-library/react";
import RecommendSlider from '../src/components/FunctionalComponents/RecommendSlider';

describe(
    'Recommend Slider Component', () => {

        beforeEach(() => {
            jest.clearAllMocks()
        })

        afterEach(() => {
            cleanup()
        })

        test('should render correctly with label above slider', () => {
            const min = 0
            const max = 100
            const step = 1
            const value = [32, 71]
            const ariaLabel = "Test Slider"
            const { getByText } = render(
                <RecommendSlider 
                 ariaLabel={ariaLabel}
                 min={min}
                 max={max}
                 step={step}
                 value={value}
                 valueLabelDisplay="auto"
                 aria-labelledby="range-slider"
                />
            )

            const recommendSlider = screen.getByTestId('recommend-slider')
            const sliderLabel = getByText(ariaLabel)
            expect(recommendSlider).toBeInTheDocument()
            expect(sliderLabel).toBeInTheDocument()
        })

        test('should match snapshot', () => {
            const tree = renderer.create(<RecommendSlider />).toJSON()
            expect(tree).toMatchSnapshot()

        })


})