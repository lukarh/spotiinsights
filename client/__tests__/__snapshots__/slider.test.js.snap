// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recommend Slider Component should match snapshot 1`] = `
<div
  className="MuiBox-root css-0"
>
  <p />
  <span
    className="MuiSlider-root MuiSlider-colorPrimary MuiSlider-sizeMedium css-187mznn-MuiSlider-root"
    data-testid="recommend-slider"
    onMouseDown={[Function]}
    style={
      {
        "color": undefined,
      }
    }
  >
    <span
      className="MuiSlider-rail css-14pt78w-MuiSlider-rail"
    />
    <span
      className="MuiSlider-track css-1gv0vcd-MuiSlider-track"
      style={
        {
          "left": "0%",
          "width": "0%",
        }
      }
    />
    <span
      className="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-eg0mwd-MuiSlider-thumb"
      data-index={0}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        {
          "left": "0%",
          "pointerEvents": undefined,
        }
      }
    >
      <input
        aria-labelledby="range-slider"
        aria-orientation="horizontal"
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={0}
        data-index={0}
        disabled={false}
        max={100}
        min={0}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        step={1}
        style={
          {
            "border": 0,
            "clip": "rect(0 0 0 0)",
            "direction": "ltr",
            "height": "100%",
            "margin": -1,
            "overflow": "hidden",
            "padding": 0,
            "position": "absolute",
            "whiteSpace": "nowrap",
            "width": "100%",
          }
        }
        type="range"
        value={0}
      />
      <span
        aria-hidden={true}
        className="MuiSlider-valueLabel css-nnid7-MuiSlider-valueLabel"
      >
        <span
          className="MuiSlider-valueLabelCircle"
        >
          <span
            className="MuiSlider-valueLabelLabel"
          >
            0
          </span>
        </span>
      </span>
    </span>
  </span>
</div>
`;
